ztp/source-crs-builder/MachineConfigChronyDynamicMaster/ptp-sync-check

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+# Evaluate PTP synchronization state and create or delete 
+# the chronyd.service condition file accordingly
+
+# The parametes below are tuned using chronyd-restart.service drop-in
+# Maximum PTP offset to be considered in-sync (ns)
+MAX_OFFSET=${MAXIMUM_WAIT_TIME:-100000}
+# PTP offset hysteresis threshold (ns)
+OFFSET_THRESHOLD=${OFFSET_THRESHOLD:-20000}
+# Maximum valid measurement age (s)
+SYNC_AGE=${SYNC_AGE:-600}
+
+# File for starting chronyd conditionally
+SYNC_MARKER_PATH="/var/run/ptp/insync"
+
+# Auxiliary logging function
+function log_debug(){
+  local LOGGER_TAG="ptp-chronyd"
+  echo "$1"
+  logger -t $LOGGER_TAG -p "syslog.debug" "$1"
+}
+
+# Compare floating point numbers
+function greater() {
+   awk -v n1="$1" -v n2="$2" 'BEGIN {printf (n1>n2?"true":"false")"\n", n1, n2}'
+}
+
+# Return absolute (for master offset)
+function abs() { 
+    echo ${1##-}
+}
+
+# Check variable is defined
+function var_defined() {
+  VAR="$1"
+  if [[ ${#VAR} -eq 0 ]]; then 
+    log_debug "Early exit at line $(caller). PTP might not be running"    
+    return 1
+  fi
+  return 0
+}
+
+# Get the last line containing masteroffset from the latest available log
+# ARGUMENTS:
+#   none
+# GLOBAL:
+#   none
+# OUTPUT:
+#   LAST_STATUS - Last line of the log containing offset and timestamp
+#     Something like "2021-10-18T16:47:06.022983665+00:00 stdout F ptp4l[8180.150]: [ens4f0] 
+#     master offset 3 s2 freq -16756169 path delay 1481"
+# RETURN:
+#   0 if all sanity checks passed, 1 otherwise
+get_last_ptp_status() {
+  LOG_SUBDIR=$(ls -tr /var/log/pods/ |grep openshift-ptp_linuxptp-daemon |tail -1)
+  if ! var_defined $LOG_SUBDIR; then
+    return 1
+  fi
+  LOG_DIR="/var/log/pods/$LOG_SUBDIR/linuxptp-daemon-container"
+  if ! var_defined $LOG_DIR; then
+    return 1
+  fi
+  LAST_LOG_STR=$(ls -ltr $LOG_DIR/ |tail -1)
+  if ! var_defined $LAST_LOG_STR; then
+    return 1
+  fi
+  LAST_LOG_ARRAY=($LAST_LOG_STR)
+  LOG_NAME=${LAST_LOG_ARRAY[-1]}
+  if ! var_defined $LOG_NAME; then
+    return 1
+  fi
+  LAST_STATUS=$(tail -2000 $LOG_DIR/$LOG_NAME |grep "] master offset" |tail -1)
```

Comment:
It might be a better calling convention to echo this string instead of setting a variable.  Then elsewhere you could do:

```
local LOG_STATUS=$(get_last_ptp_status)
```

And not have to rely on a side-effect global variable set.

Reply:
Done